lib/manager/components/transform/ReplaceFileFromString.js

```diff
@@ -0,0 +1,96 @@
+// @flow
+
+import { isEqual } from 'lodash'
+import React, {Component} from 'react'
+import {Button} from 'react-bootstrap'
+
+import type {
+  Feed,
+  FeedTransformation as FeedTransformationType
+} from '../../../types'
+
+type Props = {
+  feedSource: Feed,
+  index: number,
+  onSave: (any, number) => void,
+  transformation: FeedTransformationType
+}
+
+export function getReplaceFileFromStringValidationIssues (transformation: FeedTransformationType, feedSource: Feed): Array<string> {
+  const issues = []
+
+  // CSV data must be defined.
+  if (!transformation.csvData) {
+    issues.push('CSV data must be defined.')
+  }
+
+  return issues
+}
+
+/**
+ * Component that renders fields for ReplaceFileFromStringTransformation.
+ */
+export default class ReplaceFileFromString extends Component<Props, {csvData: ?string}> {
+  state = { csvData: null }
+
+  componentDidUpdate (prevProps: Props) {
+    const { transformation } = this.props
+    if (!isEqual(transformation, prevProps.transformation)) {
+      this.setState({ csvData: transformation.csvData })
+    }
+  }
```

Comment:
This method isn't recommended according to a [React blog post](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key). To reset some values, one recommended option is to pass in a `key` prop that will be different whenever the transformation value is different.

Reply:
Updated in https://github.com/ibi-group/datatools-ui/pull/649/commits/a7296121781a1413670f4e5ad6eb4bafabbee7bb.